client/client.go

```diff
@@ -948,6 +951,19 @@ func (c *Client) updateNodeFromFingerprint(response *cstructs.FingerprintRespons
 	if response.Resources != nil {
 		c.config.Node.Resources.Merge(response.Resources)
 	}
+
+	return c.config.Node
+}
+
+func (c *Client) updateNodeFromHealthCheck(response *cstructs.HealthCheckResponse) *structs.Node {
+	c.configLock.Lock()
+	defer c.configLock.Unlock()
+
+	// update the node with the latest driver health information
+	for name, val := range response.Drivers {
+		c.config.Node.Drivers[name] = val
+	}
+
```

Comment:
The driver health seems like a nice metric to emit. Healthy and unhealthy with driver as a label.

Reply:
Good idea. This seemed like the best option but let me know if you had something else in mind. https://github.com/armon/go-metrics/blob/2e4f2be0fe4f6b7096471aa85f2c342bff3b8f4f/metrics.go#L105